target/linux/ath79/image/nand.mk

```diff
@@ -281,6 +281,10 @@ TARGET_DEVICES += glinet_gl-x1200-nor
 
 define Device/linksys_ea4500-v3
   SOC := qca9558
+  DEVICE_COMPAT_VERSION := 1.1
+  DEVICE_COMPAT_MESSAGE := Partition table has been changed. Please \
+	install kmod-mtd-rw and erase mtd8 (syscfg) before upgrade, \
+	and use tftp to downgrade!
```

Comment:
This step seems not help (not sure). I think we need to erase `ubi` instead (dangerous) or do a new `factory` install with some tftp recovery modes.

Reply:
I have tested that erasing mtd8 (syscfg) before upgrade works, as "ubi" will automatically expand to the blank space once occupied by the former mtd8 (syscfg), with the total block number increased.

Reply:
Upgrade breaks only when the ubi inside former mtd8 (syscfg) is left after the upgrade, resulting in two ubi headers existing inside the same partition.

Reply:
> I have tested that erasing mtd8 (syscfg) before upgrade works, as "ubi" will automatically expand to the blank space once occupied by the former mtd8 (syscfg), with the total block number increased.

However, the UBIFS for rootfs_data will not automatically claim the newly available space (since it is created when mtd8 still exists, and sysupgrade does not set autoresize flag to rootfs_data). These space will be claimed during the next upgrade, when rootfs_data is removed and created again.

Reply:
> Upgrade breaks only when the ubi inside former mtd8 (syscfg) is left after the upgrade, resulting in two ubi volume tables existing inside the same ubi.

> Factory installations via either OEM firmware or tftp are confirmed working,

Factory installations via either OEM firmware works because [494-mtd-ubi-add-EOF-marker-support.patch](https://github.com/openwrt/openwrt/blob/main/target/linux/generic/pending-6.6/494-mtd-ubi-add-EOF-marker-support.patch) will erase all blocks after the EOF footer appended to the UBI image within the openwrt factory image.